angular/src/app/components/file-tree-node/file-tree-node.component.html

```diff
@@ -1,38 +1,26 @@
-<div *ngIf="node" [ngClass]="{'active': isActiveFeature()}">
-  <div *ngIf="node.isDir() && node.getPath()" class="clickable" (click)="onClick(node)">
-    <i class="fas fa-folder-open" *ngIf="displayChildren"></i>
-    <i class="fas fa-folder" *ngIf="!displayChildren"></i>
-    {{ node.getPath() }}
-  </div>
-  <div *ngIf="!node.isDir()" class=" grid-x align-justify align-middle tree-file-node" >
-    <div (click)="onClick(node)" class="grid-x cell auto clickable align-middle">
-      <app-feature-icon [feature]="node.getPayload()"></app-feature-icon>
-      {{ node.getPath() }}
+<div *ngIf="node" [ngClass]="{ active: isActiveFeature() }">
+    <div *ngIf="node.isDir() && node.getPath()" class="clickable" (click)="onClick(node)">
+        <i class="fas fa-folder-open" *ngIf="displayChildren"></i>
+        <i class="fas fa-folder" *ngIf="!displayChildren"></i>
+        {{ node.getPath() }}
     </div>
-    <div class="grid-x align-middle" *ngIf="isActiveFeature()">
-      <button class="button tiny alert hollow" (click)="delete(node)">
-        <i class="fas fa-trash"></i>
-      </button>
+    <div *ngIf="!node.isDir()" class="grid-x align-justify align-middle tree-file-node">
+        <div (click)="onClick(node)" class="grid-x cell auto clickable align-middle">
+            <app-feature-icon [feature]="node.getPayload()"></app-feature-icon>
+            {{ node.getPath() }}
+        </div>
+        <div class="grid-x align-middle" *ngIf="isActiveFeature()">
+            <button class="button tiny alert hollow" (click)="delete(node)">
+                <i class="fas fa-trash"></i>
+            </button>
+        </div>
+    </div>
+    <div class="children" *ngIf="displayChildren && node.getSize()" [style.padding-left]="node.getPath() ? '1em' : '0px'">
+        <app-file-tree-node
+            *ngFor="let child of node.getChildren(); let i = index; trackBy: i"
+            [node]="child"
+            (clickEvent)="onClickChild($event)"
+        >
+        </app-file-tree-node>
     </div>
-  </div>
-  <div class="children" *ngIf="displayChildren && node.getSize()" [style.padding-left]="node.getPath() ? '1em' : '0px'">
-  <ng-container *ngIf="node.getSize() > 500; else smallList">
-    <cdk-virtual-scroll-viewport class="scroll-viewport" itemSize="10">       
```

Comment:
i'm not following why this is dropped.

it is possible merging changes from main went bad.  we should confirm in other places that recent changes look right.